one_liners2.cc

```diff
@@ -0,0 +1,56 @@
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+#include <vector>
+#include <string>
+#include <boost/algorithm/string/classification.hpp>
+
+using namespace std;
+
+// 1. Xor to each char
+string xored_string(string s, char n) {
+    return accumulate(s.begin(), s.end(), string(""), [n](
+        const string& acc, const char& c){return acc + char(c^n);});
+}
+// reverse
+string my_reverse(string s) {
+    return accumulate(s.crbegin(), s.crend(), string(""), [](
+        const string& acc, const char& c){return acc + c;});
```

Comment:
Formally, that's a correct solution. But could you please think of a much more simple (and potentially fast) one?

Reply:
Actually, there is std::reverse, which is a one-liner and I am sure that it works faster than my one-liner, but it is probably not the purpose of this ex?

Reply:
I didn't mean this function, of course...
I guess the whole point of this exercise is to learn the rich (and some say bloated) interface of the standard library. So, just take a look at `std::string` interface :)